### PR TITLE
refactor(sdk): Change typegraph definition signature

### PR DESCRIPTION
**Pull request by Natoandro**
* Replace the expose argument with a `g` object
* Use decorator in python

### CODE REVIEW

File: typegraph/python_next/typegraph_next/graph/typegraph.py [hunk 1]
```diff
@@ -60,3 +43,81 @@ def expose(self, **kwargs: "t.func"):
         res = core.expose(store, lst, [])
         if isinstance(res, Err):
             raise Exception(res.value)
+
+
+@dataclass
+class G:
```
Comment (zifeo) [inline typegraph/python_next/typegraph_next/graph/typegraph.py:49]: Graph?

File: typegraph/python_next/typegraph_next/graph/typegraph.py [hunk 2]
```diff
@@ -60,3 +43,81 @@ def expose(self, **kwargs: "t.func"):
...
+
+
+@overload
+def typegraph(builder: Callable[[G], None]) -> Typegraph:
+    ...
+
+
+@overload
+def typegraph(
+    name: str, dynamic: Optional[bool] = None, folder: Optional[str] = None
```
Comment (zifeo) [inline typegraph/python_next/typegraph_next/graph/typegraph.py:98]: simplify with typegraph(name, {dynamic, folder} = None)?

Comment (Natoandro) [reply typegraph/python_next/typegraph_next/graph/typegraph.py:98]: I don't understand well.
Require keyword-argument??

Comment (zifeo) [reply typegraph/python_next/typegraph_next/graph/typegraph.py]: I was a bit too fast... if intelisense is confused, you could simplify with def typegraph(name: str, opts: Optional[Opts] = None) where Opts is typed dict with arguments?

Comment (Natoandro) [reply typegraph/python_next/typegraph_next/graph/typegraph.py]: It is ok now. I removed the overload so when the name should be inferred and there is no additional argument, we would have to add the parenthesis:
```python
@typegraph()
def my_tg():
    # ...
```

**Review by zifeo — DISMISSED** [review]

File: typegate/tests/e2e/typegraph/typegraphs/python/simple.py [hunk 3]
```diff
@@ -28,3 +30,6 @@
             public
         ),
     )
+
+
+test_types
```
Comment (zifeo) [inline typegate/tests/e2e/typegraph/typegraphs/python/simple.py:35]: Is that mandatory?

**Review by zifeo — APPROVED** [review]

File: typegate/tests/e2e/typegraph/typegraphs/python/simple.py (hunk 3 above)
Comment (Natoandro) [reply typegate/tests/e2e/typegraph/typegraphs/python/simple.py:35]: No. I forgot to remove.

**Review by zifeo — DISMISSED** [review]

**Review by zifeo — APPROVED** [review]

**Pull request merged**